distribution/pull_v2_unix.go

```diff
@@ -18,11 +18,22 @@ func (ld *v2LayerDescriptor) open(ctx context.Context) (distribution.ReadSeekClo
 
 func filterManifests(manifests []manifestlist.ManifestDescriptor, os string) []manifestlist.ManifestDescriptor {
 	var matches []manifestlist.ManifestDescriptor
-	for _, manifestDescriptor := range manifests {
-		if manifestDescriptor.Platform.Architecture == runtime.GOARCH && manifestDescriptor.Platform.OS == os {
-			matches = append(matches, manifestDescriptor)
+	arch := runtime.GOARCH
+	if arch == "arm" || arch == "arm64" {
+		for _, manifestDescriptor := range manifests {
+			if manifestDescriptor.Platform.Architecture == arch && manifestDescriptor.Platform.OS == os && manifestDescriptor.Platform.Variant == GOARM {
```

Comment:
This uses an exact match on `variant` with no fallback, so the failure mode here will be that images which run successfully today (although likely in a less-than-ideal way) will start failing to run with "manifest not found" errors.